frontend/lib/src/components/widgets/TextInput/TextInput.tsx

```diff
@@ -144,7 +144,20 @@ class TextInput extends React.PureComponent<Props, State> {
     // If the TextInput is *not* part of a form, we mark it dirty but don't
     // update its value in the WidgetMgr. This means that individual keypresses
     // won't trigger a script re-run.
-    this.setState({ dirty: true, value })
+    if (!isInForm(this.props.element)) {
+      this.setState({ dirty: true, value })
+      return
+    }
```

Comment:
I think it'd be a bit cleaner to change things around so that the form case is the one within the `if` statement and the "default" behavior is out of it.

It'd also be semantically clearer to have an `if`/`else` here instead of using a `return` to ensure the statements below don't run given the idea is that we always want to do one or the other